Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -67,51 +68,78 @@ static String symbolToString(const CSSValue* value)
+            auto& weight = downcast<CSSPrimitiveValue>(weightValue);
+            result.constructAndAppend(symbolToString(&symbolValue), weight.value<unsigned>());
+        } else
+            return { };
+    }
+    return result;
+}
 
-    auto& list = downcast<CSSValueList>(*pad);
-    ASSERT(list.size() == 2);
-    auto length = downcast<CSSPrimitiveValue>(list.item(0));
+static CSSCounterStyleDescriptors::Pad translatePadFromStyleProperties(const StyleProperties& properties)
+{
+    auto pad = properties.getPropertyCSSValue(CSSPropertyPad);
+    if (!pad)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(pad.get())))
+        return { };
+    ASSERT(list->size() == 2);
+    auto length = downcast<CSSPrimitiveValue>(list->item(0));
     if (!length)
         return CSSCounterStyleDescriptors::Pad();
     auto lengthValue = length->intValue();
     ASSERT(lengthValue >= 0);
-    return { static_cast<unsigned>(std::max(0, lengthValue)), symbolToString(list.item(1)) };
+    return { static_cast<unsigned>(std::max(0, lengthValue)), symbolToString(list->item(1)) };
 }
 
 static CSSCounterStyleDescriptors::NegativeSymbols translateNegativeSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative).get();
-    if (!negative || !is<CSSValueList>(negative))
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    auto negative = properties.getPropertyCSSValue(CSSPropertyNegative);
+    if (!negative)
+        return { };
 
-    auto& list = downcast<CSSValueList>(*negative);
-    ASSERT(list.size() == 1 || list.size() == 2);
     CSSCounterStyleDescriptors::NegativeSymbols result;
-    if (list.size() >= 1)
-        result.m_prefix = symbolToString(list.item(0));
-    else if (list.size() == 2)
-        result.m_suffix = symbolToString(list.item(1));
-    else {
-        ASSERT_NOT_REACHED();
-        return CSSCounterStyleDescriptors::NegativeSymbols();
+    if (auto list = dynamicDowncast<CSSValueList>(negative.get())) {
+        if (list->size() >= 1)
+            result.m_prefix = symbolToString(list->item(0));
+        else if (list->size() == 2)
+            result.m_suffix = symbolToString(list->item(1));
+        else {
+            ASSERT_NOT_REACHED();
+            return { };
+        return result;
+        }
     }
+    result.m_prefix = symbolToString(negative.get());
     return result;
 }
 
 static Vector<CSSCounterStyleDescriptors::Symbol> translateSymbolsFromStyleProperties(const StyleProperties& properties)
 {
-    auto symbolsValues = properties.getPropertyCSSValue(CSSPropertySymbols).get();
-    if (!symbolsValues || !is<CSSValueList>(symbolsValues))
+    auto symbolsValues = properties.getPropertyCSSValue(CSSPropertySymbols);
+    if (!symbolsValues)
+        return { };
+    CSSValueList* list;
+    if (!(list = downcast<CSSValueList>(symbolsValues.get())))
```

Comment:
Can we rename this variable to `symbols` ?
```suggestion
    CSSValueList* symbols = dynamicDowncast<CSSValueList>(value.get());
    if (!symbols)
```